### PR TITLE
Fix up the protostar template.js

### PR DESCRIPTION
**Pull request by okonomiyaki3000**
Pull Request for Issue # .

### Summary of Changes

Don't use unnecessary closure, just use jQuery ready function.
Use javsacript strict mode
use event delegation where possible
Make tooltips and button groups work properly with repeatable subforms

### Testing Instructions

Use protostar. 
Create a page with a repeatable subform containing a radiobutton field with the btn-group class.

### Expected result

Expect newly created rows to initialize btn-groups properly.

### Actual result

Yes, it works.

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by okonomiyaki3000** [comment]
This basically an alternative to #19722 with additional improvements and cleanup.

**Comment by ReLater** [comment]
I have tested this item :white_check_mark: successfully on b8469764c94a20b0966e105a3ee9a221e550c662<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20224">issues.joomla.org/tracker/joomla-cms/20224</a>.</sub>

File: templates/protostar/js/template.js [hunk 1]
```diff
@@ -6,47 +6,69 @@
...
-				$("label[for=" + $(this).attr('id') + "]").addClass('active btn-primary');
-			} else if ($(this).val() == 0) {
-				$("label[for=" + $(this).attr('id') + "]").addClass('active btn-danger');
-			} else {
-				$("label[for=" + $(this).attr('id') + "]").addClass('active btn-success');
+
+			$label.closest('.btn-group').find("label").removeClass('active btn-success btn-danger btn-primary');
+
+			var btnClass = 'primary';
+
```
Comment (Quy) [inline templates/protostar/js/template.js:24]: Remove extra blank line.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b8469764c94a20b0966e105a3ee9a221e550c662<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20224">issues.joomla.org/tracker/joomla-cms/20224</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20224">issues.joomla.org/tracker/joomla-cms/20224</a>.</sub>

**Pull request merged**